src/BeamAdapter/component/controller/InterventionalRadiologyController.inl

```diff
@@ -996,66 +996,45 @@ template <class DataTypes>
 void InterventionalRadiologyController<DataTypes>::sortCurvAbs(type::vector<Real> &curvAbs,
                                                                type::vector<type::vector<int> >& idInstrumentTable)
 {
-    auto newCurvAbs = sofa::helper::getWriteOnlyAccessor(d_curvAbs);
-    Real eps = d_threshold.getValue();
+    // here we sort CurvAbs   
+    std::sort(curvAbs.begin(), curvAbs.end());
 
-   // here we sort CurvAbs
-   // a buffer verctor: newCurvAbs will be filled by iteratively removing the minimal values found in CurvAbs and push_back them in newCurvAbs
-   // a threshod is used to remove the values that are "too" similars...
-   // TODO: could be improve by function "sort" ???
+    // copy threshold in variable member for std::unique
+    m_threshold = d_threshold.getValue();
 
-    newCurvAbs.clear();
-    while(curvAbs.size()>0)
-    {
-        Real xMin = 1.0e30;
-        unsigned int index_min=0;
-        for(unsigned int j=0; j<curvAbs.size(); j++)
-        {
-            if(xMin > curvAbs[j] )
-            {
-                xMin = curvAbs[j];
-               index_min=j;
-            }
-        }
-
-        type::removeIndex( curvAbs, index_min);
-
-        // verify using a eps that the same node does not already exist
-        Real xLast;
-        if (newCurvAbs.size()>0 )
-            xLast= newCurvAbs[newCurvAbs.size()-1];
-        else
-            xLast=-1.0e30;
-
-        if( fabs(xMin - xLast) > eps)
-        {
-            newCurvAbs.push_back(xMin);
-        }
-    }
+    // a threshod is used to remove the values that are "too" similars...
+    auto it = std::unique(curvAbs.begin(), curvAbs.end(), [this](const Real v1, const Real v2) {
+        return fabs(v1 - v2) < this->m_threshold;
```

Comment:
I dont see why you dont use a local variable to store threshold (and capture it in the lambda)
making a member for that seems overkill for me

Reply:
Because I didn't find how to do it :)